ui/wobbly.js

```diff
@@ -307,18 +307,34 @@ function enableWobblyFx(wm) {
         SETTINGS_HANDLER = Settings.connect('changed::wobbly-effect', actionWobblyEffectSetting);
         actionWobblyEffectSetting(Settings, 'wobbly-effect');
     });
+
+    return GLib.SOURCE_REMOVE;
+}
+
+function enableWobblyFx(wm) {
+    initAnimationsDbus(wm);
 }
 
 function disableWobblyFx(wm) {
     Settings.disconnect(SETTINGS_HANDLER);
 
     getAnimatableWindowActors().forEach(actor => {
-        actor._animatableSurface = null;
+        if (actor._animatableSurface) {
+            wm._animationsServer.unregister_surface(actor._animatableSurface);
+            actor._animatableSurface = null;
+        }
     });
 
     if (wm._wobblyEffect) {
         wm._wobblyEffect.destroy();
         wm._wobblyEffect = null;
     }
-    wm._animationsServer = null;
+    if (wm._animationsManager) {
+        wm._animationsManager.unexport();
+        wm._animationsManager = null;
+    }
+    if (wm._animationsServer) {
+        wm._animationsServer.stop(null);
```

Comment:
Maybe instead of calling `animationsServer.stop()` we could use `run_dispose()` instead - shell seems to use `run_dispose()` in several places.

Reply:
I commented in [the other PR](https://github.com/endlessm/libanimations-dbus/pull/11#discussion_r413681270) about this — I think it’s almost always a bad idea to call `run_dispose()`, since it disrupts the assumptions other code makes about `dispose()` not being called until all strong refs have been dropped.